### PR TITLE
Fix ingress IP on SPR 2.0 and namespace deletion

### PR DESCRIPTION
**Pull request by flaviodsr**
The services namespaces should be deleted after other namespaces or it can leave namespaces stuck on deletion.

This change also fixes the SPR 2.0 workflow by using a label to get the ingress IP instead.

### CODE REVIEW

File: .github/workflows/registry-2.0.yml [hunk 1]
```diff
@@ -120,8 +120,7 @@ jobs:
 
       - name: Generate helm values
         run: |
-          INGRESS_IP=$(kubectl get svc nginx-ingress-controller -n nginx-ingress \
-          -o json | jq -r ".status.loadBalancer.ingress[].ip")
+          INGRESS_IP="$(kubectl get svc --all-namespaces -l app=nginx-ingress | awk '/LoadBalancer/ { print $5 }')"
```
Comment (jsuchome) [inline .github/workflows/registry-2.0.yml:123]: So why don't you install ingress into `nginx-ingress` namespace? (sorry I did not check other parts of the code..)

What about `kubectl get svc --all-namespaces -l app=nginx-ingress -l component=controller -o jsonpath="{.items[].status.loadBalancer.ingress[].ip}"` ?

File: .github/workflows/cluster-rotate.yml [hunk 2]
```diff
@@ -98,9 +98,12 @@ jobs:
         run: |
           if [ -f "${ARTIFACT_DIR}/kubeconfig" ]; then
             export KUBECONFIG="${ARTIFACT_DIR}/kubeconfig"
-            for n in $(kubectl get ns -o=custom-columns=kube-:.metadata.name | grep -v "kube-\|default"); do
-              kubectl delete ns $n
-            done
+            kubectl get ns -o=custom-columns=kube-:.metadata.name | grep -v \
```
Comment (jsuchome) [inline .github/workflows/cluster-rotate.yml:101]: Hm, I would rather go with `--no-headers` so you don't have to create a header that you'll be ignoring with grep, just for clarity, the result is the same of course...

File: .github/workflows/cluster-rotate.yml [hunk 3]
```diff
@@ -177,8 +180,10 @@ jobs:
             esac
           done
 
-      - name: Save new CLUSTER_ID
-        run: echo "${{ steps.set_cluster_id.outputs.cluster_id }}" > CLUSTER_ID
+      - name: Save new CLUSTER_ID and set node kubeconfig
+        run: |
+          echo "${{ steps.set_cluster_id.outputs.cluster_id }}" > CLUSTER_ID
+          cp ${KUBECONFIG} ~/.kube/config
```
Comment (jsuchome) [inline .github/workflows/cluster-rotate.yml:186]: How is this change related to deleting namespaces?

File: .github/workflows/registry-2.0.yml (hunk 1 above)
Comment (flaviodsr) [reply .github/workflows/registry-2.0.yml:123]: > So why don't you install ingress into `nginx-ingress` namespace? (sorry I did not check other parts of the code..)
To install additional services on the cluster I am using the same logic as in the aks/eks workflwos where it uses the upstream nginx-ingress. However for CaaSP we use th suse nginx-ingress in that case I need a different name to differentiate it, that's why I install it on the nginx-ingress-suse namespace.
> What about `kubectl get svc --all-namespaces -l app=nginx-ingress -l component=controller -o jsonpath="{.items[].status.loadBalancer.ingress[].ip}"` ?
Well, I believe there are several ways to achieve the same result, is there any advantage on the one you mentioned?

File: .github/workflows/cluster-rotate.yml (hunk 2 above)
Comment (flaviodsr) [reply .github/workflows/cluster-rotate.yml:101]: I was not aware of the `--no-headers` I will make that change, thanks!

File: .github/workflows/cluster-rotate.yml (hunk 3 above)
Comment (flaviodsr) [reply .github/workflows/cluster-rotate.yml:186]: ops, this is something I was testing. Its just to set the kubeconfig on the node running the job. I can create another commit for it.

File: .github/workflows/registry-2.0.yml (hunk 1 above)
Comment (jsuchome) [reply .github/workflows/registry-2.0.yml:123]: Not really, I just like to achieve the result with `kubectl` alone, without the need to grep/awk the results

**Review by jsuchome — APPROVED** [review]
